Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseOpMode.java

```diff
@@ -127,37 +137,47 @@ public void driveWithIMU(double xPower, double yPower, double tPower) {
         }
 
         // calculate speed and direction of each individual motor and set power of motors to speed
-        motorFL.setPower(-yPower + xPower + tPower);
-        motorFR.setPower(-yPower - xPower - tPower);
-        motorBL.setPower(-yPower - xPower + tPower);
-        motorBR.setPower(-yPower + xPower - tPower);
+        motorFL.setPower(yPower + xPower + tPower);
+        motorFR.setPower(yPower - xPower - tPower);
+        motorBL.setPower(yPower - xPower + tPower);
+        motorBR.setPower(yPower + xPower - tPower);
     }
 
     /**
      * this method will allow the grabber to open or close given a boolean input
-     * @param isOpen true for open grabber and false for closed grabber
+     * @param position the position of the grabber
      */
-    public void driveGrabber(boolean isOpen) {
-        if (isOpen) {
-            servoGrabber.setPosition(Constants.GRABBER_OPEN_POSITION);
-        } else {
-            servoGrabber.setPosition(Constants.GRABBER_CLOSE_POSITION);
-        }
+    public void driveGrabber(double position) {
+        servoGrabber.setPosition(position);
     }
 
     /**
-     * this method will allow the slides to move upwards and downwards given a specified target position
-     * @param targetPosition target position for vertical slides motors in ticks
+     * this method will allow the slides to move to a specified target position
+     * @param targetPosition target position for slides motors in ticks
      */
     public void driveSlides(int targetPosition) {
-
-    }
-
-    /**
-     * this method will allow the turntable to turn clockwise or counterclockwise given a specified power and position
-     * @param position target position of turntable motor in ticks
-     */
-    public void driveTurntable(int position) {
-        motorTurntable.setTargetPosition(position);
+        int error = targetPosition - motorLeftSlides.getCurrentPosition();
+        double motorPower = error * 0.01;
+
+        // slides not yet at target position
+        if (Math.abs(error) > 20) {
+            // slides going down - joystick
+            if (error < 0 && error > -200) {
```

Comment:
So many magic numbers here

Reply:
fixed

Reply:
Seems like a lot of them did get fixed, but -200 still seems like a magic number to me. Probably just a few more that you missed since you had so many.

Reply:
The -200 is still there and used in multiple places in this if. That is a magic number so it should be a constant or at least explained in a comment.